src/main/java/chatbot/Storage.java

```diff
@@ -0,0 +1,168 @@
+package chatbot;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+// import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.time.LocalDate;
+// import java.time.format.DateTimeFormatter;
+// import java.time.temporal.ChronoUnit;
+import java.time.LocalTime;
+import java.util.Scanner;
+
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+/**
+* Storage class to define a location to store the data of the bot.
+*/
+public class Storage {
+    public static final String LOAD_SUCCESS = "I have successfully loaded your saved agenda, Sir.";
+    public static final String UNREADABLE_FILE = "Sorry Sir, I was unable to access the data file.";
+
+    private String filePath;
+    private String fileDirectory;
+
+    /**
+    * Class constructor.
+    *
+    * @param  filePath         String containing the desired location of the data file
+    * @param  fileDirectory    String containing the desired location of the data folder
+    */
+    public Storage(String filePath, String fileDirectory) {
+        this.filePath = filePath;
+        this.fileDirectory = fileDirectory;
+    }
+
+    /**
+    * Saves tasks in taskList into the data file duke.txt at filePath
+    * <p>
+    * The fileDirectory denotes the folder containing duke.txt
+    *
+    * @param  taskList      a TaskList containing all the tasks stored
+    * @throws DukeException throws a DukeException
+    */
+    public void saveData(TaskList taskList) throws DukeException {
+        // create a directory if it doesn't exist
+        File dataFile = new File(this.fileDirectory);
+        if (!dataFile.exists()) {
+            dataFile.mkdirs();
+        }
+
+        try {
+            FileWriter fileWriter = new FileWriter(this.filePath, false);
+            PrintWriter printWriter = new PrintWriter(fileWriter);
+            for (Task task : taskList.getTaskList()) {
+                String dataEntry = "";
+                if (task instanceof Todo) {
+                    dataEntry += "Todo";
+                    dataEntry += task.getIsDone() ? "/;DONE" : "/;NOT_DONE";
+                    dataEntry += "/;" + task.getTaskName();
+                    dataEntry += "\n";
+
+                } else if (task instanceof Deadline) {
+                    Deadline deadline = (Deadline) task;
+                    dataEntry += "Deadline";
+                    dataEntry += deadline.getIsDone() ? "/;DONE" : "/;NOT_DONE";
+                    dataEntry += "/;" + deadline.getTaskName();
+                    dataEntry += "/;" + deadline.getDate();
+                    if (deadline.getTime() == null) {
```

Comment:
A little bit of arrow heading here, you can consider using switch cases, then try encapsulating some of the other if-else statements into the various objects also